packages/api-request/src/middlewares/test/http-v1.js

```diff
@@ -0,0 +1,21 @@
+import httpV1Middleware from '../http-v1';
+
+describe( 'HTTP v1 Middleware', () => {
+	it( 'should use a POST for a PUT requests', () => {
+		const callback = ( options ) => {
+			expect( options.method ).toBe( 'POST' );
+			expect( options.headers[ 'X-HTTP-Method-Override' ] ).toBe( 'PUT' );
+		};
+
+		httpV1Middleware( { method: 'PUT', data: {} }, callback );
+	} );
+
+	it( 'shouldn\'t touch the options for GET requests', () => {
+		const requestOptions = { method: 'GET', path: '/wp/v2/posts' };
+		const callback = ( options ) => {
+			expect( options ).toEqual( requestOptions );
+		};
+
+		httpV1Middleware( requestOptions, callback );
```

Comment:
We should add [expect.hasAssertions](https://facebook.github.io/jest/docs/en/expect.html#expecthasassertions) call to both tests to make sure that `callback` was executed.